packages/e2e-tests/config/setup-screenshots.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+const fs = require( 'fs' );
+const root = process.env.GITHUB_WORKSPACE || '../../../';
+const SCREENSHOTS_PATH = root + '/screenshots';
+
+if ( ! fs.existsSync( SCREENSHOTS_PATH ) ) {
+	fs.mkdirSync( SCREENSHOTS_PATH );
+}
+
+/**
+ * @copyright Tom Esterez (@testerez) https://github.com/smooth-code/jest-puppeteer/issues/131#issuecomment-424073620
+ */
+export const registerScreenshotReporter = () => {
+	/**
+	 * jasmine reporter does not support async.
+	 * So we store the screenshot promise and wait for it before each test
+	 */
+	let screenshotPromise = Promise.resolve();
+	beforeEach( () => screenshotPromise );
+	afterAll( () => screenshotPromise );
```

Comment:
Why do we need both before and after? Couldn't we just use one of them?

Reply:
before**Each** prevents the next test from starting 
after**All** prevents jasmine from destroying the context until artifacts are stored